api/v1beta1/gcpmachine_webhook.go

```diff
@@ -99,3 +100,17 @@ func (m *GCPMachine) ValidateDelete() error {
 func (m *GCPMachine) Default() {
 	clusterlog.Info("default", "name", m.Name)
 }
+
+func validateConfidentialCompute(spec GCPMachineSpec) error {
+	if spec.ConfidentialCompute != nil && *spec.ConfidentialCompute == ConfidentialComputePolicyEnabled {
+		if spec.OnHostMaintenance == nil || *spec.OnHostMaintenance == HostMaintenancePolicyMigrate {
+			return fmt.Errorf("ConfidentialCompute require OnHostMaintenance to be set to %s, the current value is: %s", HostMaintenancePolicyTerminate, HostMaintenancePolicyMigrate)
+		}
+
+		machineSeries := strings.Split(spec.InstanceType, "-")[0]
+		if machineSeries == "" || !strings.Contains(confidentialComputeSupportedMachineSeries, machineSeries) {
```

Comment:
I'm wondering if this approach is a bit fragile? I'm not a GCP expert but on AWS for example, you have different series that are quite similar, compare an `m6` to an `m6i`, if `m6i` were in the list of supported types here, `m6` would also pass the test. Can I make a suggestion to use the [sets utils](https://pkg.go.dev/k8s.io/apimachinery/pkg/util/sets) and from there check the content from a slice rather than a CSV?

Reply:
Seems that the current version of apimachinery in capg doesn't contain the templated methods,
While type String is marked as DEPRECATED so I rather not use it.
Instead, I just validate the machine series is contained in the slice of the supported machine series.

Reply:
Look again, I just updated it

Reply:
Ahh gotcha, yeah I missed that, looks good